### PR TITLE
fix the scatternd/scatterndadd

### PR DESCRIPTION
**Pull request by ForFishes**
### PR types

Others
### PR changes

Docs
### Describe

fix the scatter_nd doc
fix the scatter_nd_add doc

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -8577,34 +8582,38 @@ def scatter_nd(index, updates, shape, name=None):
...
-    is equal to :code:`scatter_nd_add(fluid.layers.zeros(shape, updates.dtype), index, updates)` .
+    is equal to :code:`scatter_nd_add(paddle.zeros(shape, updates.dtype), index, updates)` .
     If :attr:`index` has repeated elements, then the corresponding updates are accumulated.
     Because of the numerical approximation issues, the different order of repeated elements
     in :attr:`index` may cause different results. The specific calculation method can be
     seen :code:`scatter_nd_add` . This op is the inverse of the :code:`gather_nd` op.
 
     Args:
-        index (Variable): The index input with rank > 1 and index.shape[-1] <= len(shape).
+        index (Tensor): The index input with rank > 1 and index.shape[-1] <= len(shape).
```
Comment (jzhang533) [inline python/paddle/fluid/layers/nn.py]: rank -> ndim

File: python/paddle/tensor/manipulation.py [hunk 2]
```diff
@@ -974,6 +973,78 @@ def scatter(x, index, updates, overwrite=True, name=None):
...
+def scatter_nd_add(x, index, updates, name=None):
+    """
+    **Scatter_nd_add Layer**
+
+    Output is obtained by applying sparse addition to a single value
+    or slice in a Tensor.
+
+    :attr:`x` is a Tensor with rank :math:`R`
+    and :attr:`index` is a Tensor with rank :math:`K` . Thus, :attr:`index`
+    has shape :math:`[i_0, i_1, ..., i_{K-2}, Q]` where :math:`Q \leq R` . :attr:`updates`
```
Comment (jzhang533) [inline python/paddle/tensor/manipulation.py:985]: rank的说法改成ndim吧。

File: python/paddle/tensor/manipulation.py [hunk 3]
```diff
@@ -974,6 +973,78 @@ def scatter(x, index, updates, overwrite=True, name=None):
...
+            index.shape = (2, 0)
+            updates.shape = (2, 2, 2)
+
+          we get:
+
+            output = [[67, 19], [-16, -27]]
+
+    Args:
+        x (Tensor): The x input. Its dtype should be float32, float64.
+        index (Tensor): The index input with rank > 1 and index.shape[-1] <= x.rank.
```
Comment (jzhang533) [inline python/paddle/tensor/manipulation.py]: rank改一下

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (ForFishes) [reply python/paddle/fluid/layers/nn.py]: done, thx.

File: python/paddle/tensor/manipulation.py (hunk 2 above)
Comment (ForFishes) [reply python/paddle/tensor/manipulation.py:985]: ok, i will fix it.

File: python/paddle/tensor/manipulation.py (hunk 3 above)
Comment (ForFishes) [reply python/paddle/tensor/manipulation.py]: ok, done.

**Review by jzhang533 — APPROVED** [review]
lgtm

File: python/paddle/tensor/manipulation.py [hunk 4]
```diff
@@ -974,6 +973,78 @@ def scatter(x, index, updates, overwrite=True, name=None):
...
+    Examples:
+
+        .. code-block:: python
+
+            import paddle
+            import numpy as np
+
+            x = paddle.rand(shape=[3, 5, 9, 10], dtype='float32')
+            updates = paddle.rand(shape=[3, 9, 10], dtype='float32')
+            index_data = np.array([[1, 1],
```
Comment (XiaoguangHu01) [inline python/paddle/tensor/manipulation.py:1039]: 这里不需要用numpy, 可以直接用to_tensor把list转换为tensor

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**